test/Microsoft.AspNetCore.Server.KestrelTests/MemoryPoolIteratorTests.cs

```diff
@@ -406,10 +404,9 @@ public void PeekLongAtBlockBoundarayWithMostSignificatBitsSet(int blockBytes)
             var scan = block.GetIterator();
             var originalIndex = scan.Index;
 
-            // Act
-            var result = scan.PeekLong();
-
             // Assert
+            ulong result;
+            Assert.True(scan.TryPeekLong(out result));
```

Comment:
I hadn't thought of this before, but you should add tests for the failure cases.